### PR TITLE
Detect OS automatically.

### LINKED ISSUE #18: Run detect_os automatically.
Which would result in a bit of simplification and would prevent human errors when manually building and copying the tree from one OS to another.

### PR DESCRIPTION
**Pull request by dumol**
Scope
-----

Run `./brink.sh detect_os` from `pythia` if `BUILD_ENV_VARS` file is missing.

Which would result in a bit of simplification in many places.

Changes
-------

Implemented the above which fixes #18.

**Drive-by changes** :
  - [X] Group the uploading logic in a separate script.
  - [X] The above allows showing links for testing packages after uploading.
  - [x] Simplify local `dist/` hierarchy to match uploads.
  - [x] Minor cleanups.

Testing
-------

Please review changes.

Run the automated tests.

Check links echoed for uploaded packages after a green build.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,5 +4,6 @@
 /src/*/*.tar.gz
 /src/*/*.tgz
 /src/*/*.zip
+/publish_dist_sftp_batch
```
Comment (adiroiban) [inline .gitignore:7]: ca we have this file in build-pythia/publish_dist_sftp_batch and then we will no longer need to add extra ignore?

File: functions_pythia.sh [hunk 2]
```diff
@@ -228,47 +228,45 @@ cleanup_install_dir() {
...
 
 #
 # Construct a SFTP batch file for uploading testing packages.
-# Commands prefixed with a '-' are allowed to fail.
+# Files are uploaded with a temp name and then renamed to final name.
 #
 build_publish_dist_sftp_batch() {
-    # This matches the GitHub's hierarchy for releases of production packages.
-    local upload_version_dir="$PYTHON_BUILD_VERSION.$PYTHON_PACKAGE_VERSION"
+    local full_ver="${PYTHON_BUILD_VERSION}.${PYTHON_PACKAGE_VERSION}"
```
Comment (adiroiban) [inline functions_pythia.sh:259]: Can we also have PYTHON_BUILD_VERSION and PYTHON_PACKAGE_VERSION in `BUILD_ENV_VARS` and then move all this code to ` publish_dist.sh` ?

Note that creating and then sourcing environment variables is the recommended way for passing values between GitHub Action steps... and I think that is a common practice for multi-command build system/procedures/processes 

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

I think is ok to use our own custom file as we don't depend on `$GITHUB_ENV`

**Review by adiroiban — APPROVED** [review]
All good. Thanks. Only minor comments.

File: .gitignore (hunk 1 above)
Comment (dumol) [reply .gitignore:7]: Yes, seems like a good idea, thanks!

File: functions_pythia.sh (hunk 2 above)
Comment (dumol) [reply functions_pythia.sh:259]: Have actually gone this route already. With both variables combined into a new `PYTHON_FULL_VERSION` in `BUILD_ENV_VARS`. Which also enabled shorting some extra-long lines in this repo.

But in the end I reverted to let `BUILD_ENV_VARS` be generated exclusively by the generic `brink.sh` script. The principle of least surprises and all that… I don't want the cognitive load of keeping in mind this file is generated differently in this repo compared to all the others.

But I kept the shorting of the extra-long lines though, by defining local vars where needed. E.g., `full_ver` in the snippet above. There is now no code line over 80 chars wide in the whole repo.

**Pull request merged**